version: "3.8"

services:
  backend-flask:
    image: backend-flask:latest   # <-- local image tag
    pull_policy: never
    environment:
      FRONTEND_URL: "http://localhost:3000"
      BACKEND_URL: "http://localhost:4567"
    ports:
      - "4567:4567"
    # Optional: keep this if you want live-editing from your working dir
    volumes:
      - ./backend-flask:/backend-flask
    networks:
      - internal-network
    restart: unless-stopped

  frontend-react-js:
    image: frontend-react-js:latest  # <-- local image tag
    pull_policy: never
    environment:
      REACT_APP_BACKEND_URL: "http://localhost:4567"
    ports:
      - "3000:3000"
    # Optional: remove if your image already contains the built frontend
    volumes:
      - ./frontend-react-js:/frontend-react-js
    depends_on:
      - backend-flask
    networks:
      - internal-network
    restart: unless-stopped

  db:
    image: postgres:13-alpine
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password
    ports:
      - '5432:5432'
    volumes: 
      - db:/var/lib/postgresql/data

  dynamodb-local:
    # https://stackoverflow.com/questions/67533058/persist-local-dynamodb-data-in-volumes-lack-permission-unable-to-open-databa
    # We needed to add user:root to get this working.
    user: root
    command: "-jar DynamoDBLocal.jar -sharedDb -dbPath ./data"
    image: "amazon/dynamodb-local:latest"
    container_name: dynamodb-local
    ports:
      - "8000:8000"
    volumes:
      - "./docker/dynamodb:/home/dynamodblocal/data"
    working_dir: /home/dynamodblocal

networks:
  internal-network:
    driver: bridge
    name: cruddur

volumes:
  db:
    driver: local
